src/middleware/jira-symmetric-jwt-middleware.ts

```diff
@@ -0,0 +1,130 @@
+import { decodeSymmetric, getAlgorithm } from "atlassian-jwt";
+import Logger from "bunyan";
+import { NextFunction, Request, Response } from "express";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { TokenType, verifyQsh } from "~/src/jira/util/jwt";
+import { Installation } from "~/src/models/installation";
+import { moduleUrls } from "~/src/routes/jira/atlassian-connect/jira-atlassian-connect-get";
+import { matchRouteWithPattern } from "~/src/util/match-route-with-pattern";
+
+export const jiraSymmetricJwtMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+
+	if (!await booleanFlag(BooleanFlags.NEW_JWT_VALIDATION, false)) {
+		return next();
+	}
+
+	const token = req.query?.["jwt"] || req.cookies?.["jwt"];
+
+	if (token) {
+		const issuer = getIssuer(token, req.log);
+		if (!issuer) {
+			req.log.warn("JWT claim did not contain the issuer (iss) claim");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const installation = await Installation.getForClientKey(issuer);
+		if (!installation) {
+			req.log.warn("No Installation found");
+			return res.status(401).send("Unauthorised");
+		}
+
+		const secret = await installation.decrypt("encryptedSharedSecret");
+
+		const tokenType = checkPathValidity(req.originalUrl) ? TokenType.normal : TokenType.context;
+		try {
+			verifySymmetricJwt(token, secret, req, tokenType, req.log);
+		} catch (err) {
+			req.log.warn({ err }, "Could not verify symmetric JWT");
+			return res.status(401).send("Unauthorised");
+		}
+
+		res.locals.installation = installation;
+		res.locals.jiraHost = installation.jiraHost;
+		req.session.jiraHost = installation.jiraHost;
+
+		if (req.cookies.jwt) {
+			res.clearCookie("jwt");
+		}
+		return next();
+
+	} else if (req.session?.jiraHost) {
+
+		const installation = await Installation.getForHost(req.session.jiraHost);
+		if (!installation) {
+			req.log.warn("No Installation found");
+			return res.status(401).send("Unauthorised");
+		}
+
+		res.locals.installation = installation;
+		res.locals.jiraHost = installation.jiraHost;
+		return next();
+	}
+
+	req.log.info("No Token found");
```

Comment:
1. Should it be `warn`? 
2. Change text to `No token found and session cookie has not jiraHost`?